### ISSUE TITLE
improve angularX component generation for side-by-side

### ISSUE THREAD

**Issue by xetys** [jhipster/generator-jhipster#8839]
##### **Overview of the feature request**

Like in #8509 the generated files for angular components have unnecessary private methods, which force a user doing side-by-side either reimplement those methods with a leading "_" in the name or change the actual code, what we don't want if doing SBS.

 ##### **Motivation for or Use Case**

Like @agoncal, I try to work using the side-by-side approach in my projects. Changing the private methods at least to protected, will make extending components looking really good in angular.

 ##### **Suggest a solution**

Methods like onError, onSaveError, but also the injected dependencies should be at least protected, so

- the user can use injected services (or inject own extensions) down to the root components functionality. You simply don't use `private` or `protected` keywords inside constructor (but as parameters for super()).
- extended methods like `subscribeToSaveResponse` can use onError, without reimplementing in

Here is an example of how a side-by-side extended component looks like, if this feature request would be implented:

``` typescript
export class TrackUpdateExtendComponent extends TrackUpdateComponent implements OnInit {

    constructor(
        dataUtils: JhiDataUtils,
        jhiAlertService: JhiAlertService,
        trackService: TrackService,
        songService: SongService,
        activatedRoute: ActivatedRoute
    ) {
        super(dataUtils, jhiAlertService, trackService, songService, activatedRoute);
    }

    ngOnInit() {
        super.ngOnInit();
    }

    protected subscribeToSaveResponse(result: Observable<HttpResponse<ITrack>>) {
      super(result);
      if (something()) {
        this.onError('something went wrong');
      }
    }
}
```

Currently, I couldn't override `subscribeToSaveResponse`, so I would need to make a `_subscribeToSaveResponse`, override `save`, use '_' as prefix in all required services etc.

##### **Related issues or PR**
#8509 

-   [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by deepu105** [comment]
Can you update it plz. I guess they can be protected

**Comment by xetys** [comment]
I did the update at least on the entity files, but doing that I faced the question "What about the non-entity client code? What should be kept private, what can be switched to protected?"

So, after a bit of reviewing the use-cases for `private` vs `protected` I can summarize: `private` is better when we explicitly don't want our users to inherit those fields. Protected fields are semi-public, as a user can inherit this fields. That helps to do side-by-side a lot. But on the other side, changes of the base classes during JHipster upgrade can break the extended classes. That's why it's "semi-public".

So doing private makes these fields more save, but forces the user to completely reimplement copies of these fields during side-by-side. That prevents breaking changes, but also important updates, and ends up in more code.

Doing protected dramatically increases side-by-side experience, without touching any generated code, and also the risk of breaking changes.

IMHO, giving the user a tool, allowing to edit less generated code than before, decreases the damage after JHipster upgrades much more, then doing protected fields increases break risks.

However, a quick search pointed out we are using 184 private fields in total in our client generation, so switching all of them to `protected` is possibly a huge change I would like to discuss with the @jhipster/developers and also @agoncal.


WDYT?
